Source/DafnyRuntime/DafnyRuntimeRust/src/tests/mod.rs

```diff
@@ -71,7 +71,51 @@ mod tests {
         }
     }
 
+    // This one is still the bad kind of recursive
     #[test]
+    fn test_sequence_right() {
+        let a: Sequence<i32> = Sequence::<i32>::from_array_owned(vec![1, 2]);
+        let b = Sequence::<i32>::from_array_owned(vec![3, 4]);
+        let c = Sequence::<i32>::from_array_owned(vec![5, 6]);
+        let d = Sequence::<i32>::from_array_owned(vec![7, 8]);
+        let e = Sequence::<i32>::from_array_owned(vec![9, 10]);
+        let f = Sequence::<i32>::from_array_owned(vec![11, 12]);
+        let g = Sequence::<i32>::from_array_owned(vec![13, 14]);
+        let h = Sequence::<i32>::from_array_owned(vec![15, 16]);
+        let c1 = Sequence::<i32>::new_concat_sequence(&a, &b);
+        let c2 = Sequence::<i32>::new_concat_sequence(&c1, &c);
+        let c3 = Sequence::<i32>::new_concat_sequence(&c2, &d);
+        let c4 = Sequence::<i32>::new_concat_sequence(&c3, &e);
+        let c5 = Sequence::<i32>::new_concat_sequence(&c4, &f);
+        let c6 = Sequence::<i32>::new_concat_sequence(&c5, &g);
+        let c7 = Sequence::<i32>::new_concat_sequence(&c6, &h);
+        let arr = c7.to_array();
+        assert_eq!(*arr, vec![1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16]);
+    }
+
+    // This one is successfully tail recursive
+    #[test]
+    fn test_sequence_left() {
+        let a: Sequence<i32> = Sequence::<i32>::from_array_owned(vec![1, 2]);
+        let b = Sequence::<i32>::from_array_owned(vec![3, 4]);
```

Comment:
```suggestion
        let b = seq![3, 4];
```
I would prefer to see this way of defining sequences in this file because it's shorter. Also, if you could update all of them, that's be even better.

Reply:
Done